internal/controllers/machine/machine_controller_status.go

```diff
@@ -66,12 +70,12 @@ func (r *Reconciler) reconcileStatus(ctx context.Context, s *scope) {
 
 	// TODO: Set the uptodate condition for standalone pods
 
+	setDeletingCondition(ctx, s.machine, s.reconcileDeleteExecuted, s.deletingReason, s.deletingMessage)
```

Comment:
Stefan Instead of having scope variables that mutate through reconciliation have you considered putting the checkConditionStatus logic in a centralized single place? That might result in some code/execution duplication but it would make easy to grasp and maintain status. On the other side it might not always be practical so just curious if you thought about it

Reply:
Do you mean putting all the code needed to calculate the condition into a single place? The problem is that we need a huge part of our controller code to caculate the conditions. So we either would have to duplicate these parts or move them all into this single place

(I might have misunderstood what you're suggesting though)

Reply:
Yes, that's what I meant. It'd be kind of inverting the loop logic, e.g. First calculate status by observing most recent state of the world (querying the world). Then reconcile intent and use the observed state to determine course of actions if needed. But anyways as mentioned it might be impractical and I don't want to hijack this PR on this discussion :) just something to think about.